apis/Google.Cloud.Firestore.Data/Google.Cloud.Firestore.Data/WriteBatch.cs

```diff
@@ -88,6 +88,34 @@ public WriteBatch Delete(DocumentReference documentReference, Precondition preco
         }
 
         /// <summary>
+        /// Adds an update operation that updates just the specified fields paths in the document, with the corresponding values.
+        /// </summary>
+        /// <param name="documentReference">A document reference indicating the path of the document to update. Must not be null.</param>
+        /// <param name="updates">The updates to perform on the document, keyed by the field path to update. Fields not present in this dictionary are not updated. Must not be null.</param>
+        /// <param name="precondition">Optional precondition for deletion. May be null, which is equivalent to <see cref="Precondition.MustExist"/>.</param>
+        /// <returns>This batch, for the purposes of method chaining.</returns>
+        public WriteBatch Update(DocumentReference documentReference, IDictionary<FieldPath, object> updates, Precondition precondition = null)
+        {
+            var serializedUpdates = updates.ToDictionary(pair => pair.Key, pair => ValueSerializer.Serialize(pair.Value));
+            var deconstructed = ExpandObject(serializedUpdates);
+
+            // TODO: Validate that the precondition is reasonable (i.e. not "MustNotExist")?
+            var write = new Write
+            {
+                CurrentDocument = (precondition ?? Precondition.MustExist).Proto,
+                Update = new Document
+                {
+                    Fields = { deconstructed },
+                    Name = documentReference.Path,
+                },
+                UpdateMask = new DocumentMask { FieldPaths = { updates.Keys.Select(fp => fp.EncodedPath) } }
```

Comment:
Might be better to read from `serializedUpdates`, rather than `updates`, just in case `updates` should only be iterated over once (unlikely, but possible...).

Reply:
Good plan, will do. (Previously the parameter was just `Dictionary<FieldPath, object>` which would have made this more reasonable.)